app/core/routeObject.js

```diff
@@ -229,10 +239,34 @@ function (FauxtonAPI, Backbone) {
     },
 
     setView: function (selector, view) {
+      this.removeView(selector);
+      this.removeComponent(selector);
       this.views[selector] = view;
       return view;
     },
 
+    setComponent: function (selector, component) {
+      this.removeView(selector);
+      this.removeComponent(selector);
+      this.reactComponents[selector] = component;
+    },
+
+    removeComponent: function (selector) {
+      if (_.has(this.reactComponents, selector)) {
+        React.unmountComponentAtNode($(selector)[0]);
+        this.reactComponents[selector] = null;
+        delete this.reactComponents[selector];
+      }
+    },
+
+    removeComponents: function () {
+      _.each(this.reactComponents, function (component, selector) {
+        this.removeComponent(selector);
+      }, this);
+
+      this.reactComponents = {};
+    },
+
```

Comment:
can you write a few unit tests for those methods?

You will probably need https://github.com/apache/couchdb-fauxton/pull/356 before, going to merge it now